src/shx.js

```diff
@@ -19,36 +18,28 @@ var CMD_BLACKLIST = [
   'exec',
 ];
 
-function shx(argv) {
-  if (argv.length < 3) {
+const shx = (argv) => {
+  const [fnName, ...args] = argv.slice(2);
+  if (!fnName) {
     console.error('Error: Missing ShellJS command name');
     return EXIT_CODES.SHX_ERROR;
   }
 
-  var cmds = parseArgs(argv.slice(2));
-
-  // If something fails, we want to execute all the commands, but still error at the end.
-  var failed = false;
-
-  for (var i = 0; i < cmds.length; ++i) {
-    if (!cmds[i]) continue; // If you did foo;bar;baz;
-    var cmd = shell[cmds[i][0]];
-    if (!cmd || typeof cmd !== 'function') {
-      console.error('Error: Invalid ShellJS command: ' + cmd);
-      return EXIT_CODES.SHX_ERROR;
-    } else if (CMD_BLACKLIST.indexOf(cmd) > -1) {
-      console.error('Warning: shx ' + cmd + ' is not supported');
-      return EXIT_CODES.SHX_ERROR;
-    }
-
-    var ret = cmd.apply(shell, cmds[i].slice(1));
-    if (cmd.name !== '_echo') {
-      printCmdRet(ret);
-    }
-    if (shell.error() || (ret && ret.code && ret.code !== 0)) failed = true;
+  // validate command
+  if (typeof shell[fnName] !== 'function') {
+    console.error(`Error: Invalid ShellJS command: ${fnName}.`);
+    return EXIT_CODES.SHX_ERROR;
+  } else if (CMD_BLACKLIST.includes(fnName)) {
+    console.error(`Warning: shx ${fnName} is not supported`);
+    return EXIT_CODES.SHX_ERROR;
   }
 
-  return failed ? EXIT_CODES.CMD_FAILED : EXIT_CODES.SUCCESS;
-}
+  const ret = shell[fnName](...args);
+
+  // echo already prints
+  if (fnName !== 'echo') printCmdRet(ret);
+
+  return shell.error() ? EXIT_CODES.CMD_FAILED : EXIT_CODES.SUCCESS;
```

Comment:
Since only `exec` returns a code (AFAICT) and it is blacklisted, we don't need to handle the `code` property.